tests/publiccloud/registercloudguest.pm

```diff
@@ -0,0 +1,79 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cloud-regionsrv-client
+# Summary: Test system (re)registration
+# https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+#
+# Maintainer: <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use version_utils;
+use registration;
+use warnings;
+use testapi;
+use strict;
+use utils;
+use publiccloud::utils;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Preserve args for post_fail_hook
+    $self->{instance} = $args->{my_instance};
+
+    # Create $instance to make the code easier to read
+    my $instance = $args->{my_instance};
+
+    my $regcode = (is_byos()) ? get_required_var('SCC_REGCODE') : undef;
+    my $regcode_param = ($regcode) ? "-r $regcode" : '';
```

Comment:
```suggestion
    my $regcode_param = (is_byos()) ? "-r " . get_required_var('SCC_REGCODE')  : '';
```

if I haven't miss something variable `$regcode` used only here , so it can be dropped